terraform/charm/main.tf

```diff
@@ -0,0 +1,91 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+#--------------------------------------------------------
+# 1. DEPLOYMENTS
+#--------------------------------------------------------
+resource "juju_application" "etcd" {
+  charm {
+    name     = "charmed-etcd"
+    channel  = var.channel
+    revision = var.revision
+    base     = var.base
+  }
+  config             = var.config
+  model              = var.model
+  name               = var.app_name
+  units              = var.units
+  constraints        = var.constraints
+  storage_directives = var.storage
+
+  dynamic "expose" {
+    for_each = var.expose ? [1] : []
+    content {}
+  }
+
+  endpoint_bindings = [
+    for k, v in var.endpoint_bindings : {
+      endpoint = k, space = v
+    }
+  ]
+}
+
+resource "juju_application" "self-signed-certificates" {
```

Comment:
unlike opensearch, this is not a mandatory requirement for the etcd charm module
consider introducing a conditional such as: https://github.com/canonical/opensearch-dashboards-operator/blob/2/edge/terraform/main.tf#L35-L36

As well as in all below subsequent references to the tls operator